device/transport/amqp/src/amqp.ts

```diff
@@ -419,25 +437,37 @@ export class Amqp extends EventEmitter implements DeviceTransport {
           disableTwinDesiredPropertiesUpdates: (callback) => this._twinClient.disableTwinDesiredPropertiesUpdates(handleResult('could not disable twin desired properties updates', callback)),
           enableC2D: (callback) => {
             /*Codes_SRS_NODE_DEVICE_AMQP_41_003: [The `enableC2D` method shall attach the C2D link only if it is not already attached.] */
-            if (!this._c2dLink) {
-              debug('attaching C2D link');
-              this._amqp.attachReceiverLink(this._c2dEndpoint, null, (err, receiverLink) => {
+            if (this._c2dLink) {
+              debug('C2D link already attached, doing nothing....');
+              process.nextTick(callback);
+            } else {
+              debug('waiting for a C2D link');
+              this._amqpLinkEmitter.once('receiverLinkAttached', (err) => {
```

Comment:
You should create a Event Emitter specific for setting up the c2d receiver.  If you share and a c2d enabled at the same time you are setting up the sender, you won't actually set up the c2d receiver.